glusterd2/volume/volume-utils.go

```diff
@@ -199,3 +200,54 @@ func CreateVolumeInfoResp(v *Volinfo) *api.VolumeInfo {
 		SnapList:  v.SnapList,
 	}
 }
+
+// MountLocalBricks mounts bricks of auto provisioned volumes
+func MountLocalBricks() error {
+	volumes, e := GetVolumes()
+	if e != nil || volumes == nil {
+		// In case cluster doesn't have any volumes configured yet,
+		// treat this as success
+		log.Debug("Failed to retrieve volumes")
+		return nil
+	}
+
+	// Get list of mounted dirs
+	out, err := utils.ExecuteCommandOutput("df", "--output=target")
+	if err != nil {
+		log.WithError(err).Error("failed to get list of mounts")
+		return err
+	}
+
+	mounts := make(map[string]struct{})
+	mountdirs := strings.Split(string(out), "\n")
+
+	if len(mountdirs) > 1 {
+		// Skip the header line from df output
+		for _, md := range mountdirs[1:] {
+			mounts[md] = struct{}{}
+		}
+	}
+
+	for _, v := range volumes {
+		for _, b := range v.GetLocalBricks() {
+			// Mount all local Bricks if they are auto provisioned
+			if b.MountInfo.DevicePath != "" {
+				if _, exists := mounts[b.MountInfo.Mountdir]; exists {
```

Comment:
MountInfo.Mountdir actually contains the directory after mount root, for example let's say mount point is /bricks, but brick path could be /bricks/b1 . in this case MountDir will have /b1. 

The name is bit consfusing, I just took the same name from gd1.

Reply:
I added without that directory. See Path and Mountdir here https://github.com/gluster/glusterd2/pull/973/files#diff-3e14b73c75a337a5a43f9719dcd68228R131

Reply:
While I agree that Mountdir is not an optimal name, storing both path and actual mount dir seems redundant to me. In most case, path will be equal to mount root. And in almost all cases, directories after mount root will be less in length.
So I think we should store path and directory after mount root, and creates mount root using these two.

Reply:
makes sense, we can make that optimization. That can be separate PR since that needs to be handled in many places including response. `brick.Path = brick.Mountdir + brick.Dir`

Reply:
That's fine for me. We should store path and dir and calculate mount root like MountRoot := b.Path[:len(b.Dir)] .

One more thing As soon as this patch get's in, the snapshot functionality will break. Because, if you have any restored volume or cloned volume, then Mountdir is different in both case.

Reply:
If e2e tests don't fail then I prefer submitting snapshot related changes as separate PR. is that okay?

Reply:
sure. We can create an issue